src/App.tsx

```diff
@@ -1,24 +1,207 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useState,
+  useEffect,
+  useMemo,
+  useCallback,
+} from 'react';
+import { TodoList } from './component/TodoList/todoList';
+import { Footer } from './component/Footer/footer';
+import { Header } from './component/Header/header';
+import { Todo } from './type/Todo';
+import { GetFilter } from './type/GetFilter';
+import { getTodosPost, deleteTodo, getTodos } from './api/todos';
+import { client } from './utils/fetchClient';
+import { NewTodo } from './type/NewTodo';
+import { ErrorMessage } from './type/error';
+import { ErrorModal } from './component/ErrorModal/errorModal';
 
-const USER_ID = 0;
+const USER_ID = 10570;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = React.useState('');
+  const [querySearch, setQuerySearch] = useState('');
+  const [filterBy, setFilterBy] = useState<GetFilter>(GetFilter.ALL);
+
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const filteredTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (filterBy) {
+        case GetFilter.ACTIVE:
```

Comment:
filter each case separately. it will avoid redundant filter case GetFilterAll. in this or default case just return todos